crates/storage/provider/src/providers/database/mod.rs

```diff
@@ -518,17 +513,13 @@ mod tests {
 
     #[test]
     fn common_history_provider() {
-        let chain_spec = ChainSpecBuilder::mainnet().build();
-        let db = create_test_rw_db();
-        let provider = ProviderFactory::new(db, Arc::new(chain_spec));
-        let _ = provider.latest();
+        let factory = create_test_provider_factory();
+        let _ = factory.latest();
     }
```

Comment:
what does this test do?

Reply:
it's a leftover checking the lifetime bounds of state provider, should be ok to remove in another PR